tests/foreman/cli/test_repository.py

```diff
@@ -691,3 +699,162 @@ def test_positive_upload_content(self):
             "Successfully uploaded file '{0}'".format(RPM_TO_UPLOAD),
             result[0]['message'],
         )
+
+
+class RepositoryExportTestCase(CLITestCase):
+    """Tests for exporting a repository via CLI"""
+
+    def setUp(self):
+        """Create a directory for export, configure permissions and satellite
+        settings
+        """
+        self.export_dir = gen_string('alphanumeric')
+
+        # Create a new 'export' directory on the Satellite system
+        result = ssh.command('mkdir /mnt/{0}'.format(self.export_dir))
+        self.assertEqual(result.return_code, 0)
+
+        result = ssh.command(
+            'chown foreman.foreman /mnt/{0}'.format(self.export_dir))
+        self.assertEqual(result.return_code, 0)
+
+        result = ssh.command('ls -Z /mnt/ | grep {0}'.format(self.export_dir))
+        self.assertEqual(result.return_code, 0)
+        self.assertGreaterEqual(len(result.stdout), 1)
+        self.assertIn('unconfined_u:object_r:mnt_t:s0', result.stdout[0])
+
+        # Fix SELinux policy for new directory
+        result = ssh.command(
+            'semanage fcontext -a -t foreman_var_run_t "/mnt/{0}(/.*)?"'
+            .format(self.export_dir)
+        )
+        self.assertEqual(result.return_code, 0)
+
+        result = ssh.command('restorecon -Rv /mnt/{0}'.format(self.export_dir))
+        self.assertEqual(result.return_code, 0)
+
+        # Assert that we have the correct policy
+        result = ssh.command('ls -Z /mnt/ | grep {0}'.format(self.export_dir))
+        self.assertEqual(result.return_code, 0)
+        self.assertGreaterEqual(len(result.stdout), 1)
+        self.assertIn(
+            'unconfined_u:object_r:foreman_var_run_t:s0', result.stdout[0])
+
+        # Update the 'pulp_export_destination' settings to new directory
+        Settings.set({
+            'name': 'pulp_export_destination',
+            'value': '/mnt/{0}'.format(self.export_dir),
+        })
+
+    def tearDown(self):
```

Comment:
The tearDown will not be executed if setUp fails and since you are doing assertions on the setUp it can fail after creating the directory and the clean up will not happen.

Use `addCleanup` instead. For more info check https://docs.python.org/2/library/unittest.html#unittest.TestCase.addCleanup